drm_hwcomposer: Various drm_hwcomposer improvements / HWC 2.3

arminn opened the pull request:
Fix GetDisplayIdentificationData, introduced
by commit: b3d817815fad.
Edid data should be copied instead of assign.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
drm_hwcomposer: Fix GetDisplayIdentificationData

Fix GetDisplayIdentificationData, introduced
by commit: b3d817815fad.
EDID data should be copied instead of been assigned.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym commented:
Few proposals for commit message only. Patch itself looks great.

- I would suggest to use 'assigned' instead of 'assign' to align it with 'copied'.
- Maybe it will be better to use EDID (all caps) as it is abbreviation.

rshym commented:
LGTM

arminn committed:
drm_hwcomposer: Propagate PLATFORM_SDK_VERSION to cflags

PLATFORM_SDK_VERSION is needed for already present
code (GetDisplayIdentificationData/GetDisplayCapabilities),
which implements some of the android.hardware.graphics.composer@2.3 API.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
drm_hwcomposer: Implement GetRenderIntents stub

Fix VTS test GraphicsComposerHidlTest.GetRenderIntentsBadDisplay.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
drm_hwcomposer: Implement SetColorModeWithIntent stub

Fix VTS test GraphicsComposerHidlTest.SetColorMode_2_2BadParameter.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

arminn committed:
[HACK] drm_hwcomposer: EDID SIMULATION

Will be used prior EDID implementation in pvDisplay.

Signed-off-by: Andrii Chepurnyi <andrii_chepurnyi@epam.com>

rshym commented on drmhwctwo.cpp:
```diff
@@ -991,21 +991,57 @@ HWC2::Error DrmHwcTwo::HwcDisplay::ValidateDisplay(uint32_t *num_types,
     uint8_t *outPort, uint32_t *outDataSize, uint8_t *outData) {
   supported(__func__);
 
+
```
Seems like it's inserted by mistake.

rshym commented on drmhwctwo.cpp:
```diff
@@ -1016,6 +1052,7 @@ HWC2::Error DrmHwcTwo::HwcDisplay::GetDisplayIdentificationData(
   *outPort = connector_->id();
 
   return HWC2::Error::None;
+
```
Seems like it's inserted by mistake.

rshym commented on drmhwctwo.cpp:
```diff
@@ -991,21 +991,57 @@ HWC2::Error DrmHwcTwo::HwcDisplay::ValidateDisplay(uint32_t *num_types,
+    ALOGE("$$$ Mapped [%d] -> [%d] for edid", id, blob_id);
+  }
+  blob = drmModeGetPropertyBlob(drm_->fd(), connector_to_id[id]);
+#endif
```
Maybe it's good to append `/* SIMULATE_EDID */`
